tgui/packages/tgui/interfaces/Sleeper.js

```diff
@@ -0,0 +1,183 @@
+/**
+ * @file
+ * @copyright 2020
+ * @author BenLubar (https://github.com/BenLubar)
+ * @license ISC
+ */
+
+import { useBackend } from "../backend";
+import { Box, Button, Section, Knob, ProgressBar, LabeledList, Flex, Icon, TimeDisplay, HealthStat } from "../components";
+import { Window } from "../layouts";
+import { formatTime } from "../format";
+
+const damageNum = num => num <= 0 ? "0" : num.toFixed(1);
+
+export const Sleeper = (props, context) => {
+  const { data, act } = useBackend(context);
+  const {
+    sleeperGone,
+    hasOccupant,
+    occupantStat,
+    health,
+    oxyDamage,
+    toxDamage,
+    burnDamage,
+    bruteDamage,
+    recharging,
+    rejuvinators,
+    isTiming,
+    time,
+    timeStarted,
+    timeNow,
+  } = data;
+
+  const curTime = Math.max(timeStarted
+    ? (time || 0) + timeStarted - timeNow
+    : (time || 0), 0);
+  const canInject = hasOccupant && !isTiming && !recharging && occupantStat < 2;
+
+  return (
+    <Window
+      theme="ntos"
+      width={440}
+      height={440}>
+      <Window.Content>
+        <Section title="Occupant Statistics"
+          buttons={
+            <Button
+              icon="eject"
+              align="center"
+              color="good"
+              disabled={!hasOccupant || !!isTiming}
+              onClick={() => act("eject")}>
+              {"Eject"}
+            </Button>
+          }>
+          {!hasOccupant && (sleeperGone ? "Check connection to sleeper pod." : "The sleeper is unoccupied.")}
+          {!!hasOccupant && (
+            <LabeledList>
+              <LabeledList.Item label="Status">
+                <Icon color={occupantStat > 1 ? "bad" : occupantStat === 1 ? "average" : "good"}
+                  name={occupantStat === 0 ? "check" : occupantStat === 1 ? "bed" : "skull"} />
+                {
+                  occupantStat === 0 ? " Conscious"
+                    : occupantStat === 1 ? " Unconscious"
+                      : occupantStat === 2 ? " Dead"
+                        : (" Error " + occupantStat)
+                }
+              </LabeledList.Item>
+              <LabeledList.Item label="Overall Health">
+                <ProgressBar
+                  value={health}
+                  maxValue={100}
+                  minValue={0}
+                  ranges={{
+                    good: [90, Infinity],
+                    average: [50, 90],
+                    bad: [-Infinity, 50],
+                  }} />
```

Comment:
I'd want to double check how this implementation works for people with lower max health than 100 (e.g. ~~victims~~ patients of multiple clone attempts).

Consistency with how handheld health analyzers work is probably the best way to go, can you confirm this is the case here and/or tweak if needed?

Reply:
Looks like it doesn't care about actual health, it only cares about percentage.